### PR TITLE
Balance changes to swords, energy shields and modsuit shields.

### PR DESCRIPTION
**Pull request by necromanceranne**
### Sword Weaponry

Mundane sword weapons of all sorts do not block ``LEAP_ATTACK`` attacks whatsoever. These attacks include tackles, xeno tackles and bodythrows.

Energy swords and double energy swords only gain 25% block probability against such attacks.

### Double Energy Sword

No longer grants outright energy projectile immunity while employed. Instead, it just has a high probability of reflecting (the typical 75% to block any other attack). So, very solid defense against energy projectiles, but not immunity.

Against non-reflectable projectiles, like ballistics or nanite bullets, the desword only has 50% block, similar to an energy sword.

To compensate for the loss of defensive power, we'll make it all the more rewarding for getting on top of someone with the sword by giving it 40 force while active. And also it costs 13 TC.

### Combat Energy Shield

This also lost outright energy projectile immunity, but gained the standard blocking power of shields on top of the ability to reflect energy projectiles when they block them. This significantly increases the shields potential effectiveness while no longer pigeonholing the shield to only energy weapons. (This makes them exceptionally good against tackles and body throws, by the by).

Deathsquads still have the perfect deflection energy shield so that they can continue to spam pulse shots with impunity.

### MODsuit Shield Module

Only has one charge instead of three, but it recharges in half the time. This is no longer such a perfect defense, and does somewhat need you to be thinking about how you're utilizing the shield rather than not thinking about defense at all by barreling forward under three potential hits worth of protection.

Also much cheaper, at almost half price of 8 TC. Because of how cheap it is (and how much it still is necessary to keep you alive), I've put it into the core equipment box (which brings the price up to 22 TC. As a reminder, this is not meant to be at any discount, and is more aimed towards teaching newer players which items contribute towards success. If you don't want all the times within, don't buy this box, just buy what you want separately.)

## Why It's Good For The Game

This is a doozy of an explanation, I hope you're ready for it under the spoiler.
<details>
With my tackling and bodythrow prs, numerous people expressed exasperation at the fact that these two tools may have been keeping some outlier antagonist gear from becoming too easy to steamroll with if you already knew what you were doing. My intent was to create consistent rules and behaviours that both A) did not rely on bugs to keep the balance of power from tipping one way or the other, and B) was at least consistent or had consistent rules established.

This PR is tackling overperforming gear combinations for already competent nukies that may have, over time, crept out of control, and applying some consistency to the rules around similar equipment. 

AND also deals with quite possibly the most braindead element of game design we've tolerated for about a decade, and half a decade after it was necessary to maintain that decision.

Part of the culprit of this issue is that, specifically in regards to nukies, crew can't use the vast majority of their weapons effectively against them. This largely is because this antagonist can gain immunities to those types of equipment. And that is rapidly increasing as we move closer towards outright ballistic removal. I don't think the game is made healthier by everyone on the station having to fight armed mercenaries with spears, and doesn't make much thematic sense either. More so, most greener players probably just don't know this is how it works, and so surprise Pikachu when their lasers bounce off nukies harmlessly. (This bit reminds me of the problem of new players using disablers against simple mobs)

But of course, that isn't the only part of the problem. The other half is due to being able to be layered on a much more broad defensive tool in the form of the MODsuit Shield Module, whose three charges could render the mindful nukie near untouchable if they're pairing it with some other layered defense, such as a desword. Notice that this doesn't really address armor. The culprit is negation, and not mitigation, and we should be sparing in how easily we hand out outright effect negation simply because it isn't super obvious to a new player why it happened, and how to resolve it. At the very least, we should look to find ways to add options for players to overcome these problems. Especially with teamwork.

Energy projectile immunity made sense while there floated around an energy projectile that ostensibly would down you in a single shot. Nukies ALSO had projectile weapons that worked much the same (c-20r stun bullets, taser shot bulldogs, etc.), so it was predominantly tit-for-tat. These immunity granting equipment pieces forced crew members to get shotguns and ballistic guns to fight these dangers; something more available at the time.

We've exercised large bits and pieces of this from the game a long time ago, but we still have some remnants convinced we're still in a taser-rich, ballistic available environment. We need to move the games languishing tools into the modern era and re-established their place in the game. Namely, the double-energy sword and the combat energy shield are almost entirely unchanged besides refactors for the last decade or so, even while the game around them have changed. They've been a continuous sore point for me in all my time developing and a constant nagging issue. I want to deal with it now.

MODsuit Shield Module is just kind of really good and only made stronger the more defenses you have. It's good to have a defense like this, but I think it is too brain dead. With only one charge, it will save you from a lost joust here and there, but it won't make it as simple as running right at every problem you encounter and eating a volley of attacks while you kill someone with impunity.

**With regards to traitors**, since they also get double-energy swords; I'm open to suggestions if this is hitting them far too hard, but I'm not terribly concerned using this weapon for a few reasons. **Firstly**, I think their presence amongst the crew make it a much better weapon for tots than nukies (in isolation) simply because they can find ways to exploit it via tools they gather from the station. It is a force multiplier. Traitors also have a much bigger element of surprise usually. **Secondly**, round-start traitors typically grow to be a bit stronger over time, but I don't foresee many waiting to pay for the double-energy sword unless they're already flush with TC. So if a traitor is in a position after they've unlocked access to it to buy one of these, they are probably doing pretty okay for themselves.
</details>

### TL;DR

Defense stacking and attack immunities are not particularly healthy things to both design around, or experience in-game. They are kind of just relics of the past made only sorer once I ripped off a few bandaids. This is a source of a number of symptomatic issues in the game, so let's fix that and make it easier on all of us going forward.

Much of the way these things worked operated on extremely outdated design considerations. It doesn't make sense for them to work like this today, and only makes things harder by keeping the status quo.

## Changelog
:cl:
balance: Mundane sword-like and medieval weapons are not able to block tackles, xenomorph tackles and body throws.
balance: The double-energy sword and energy sword have trouble blocking physical projectiles, body throws and tackles.
balance: The double-energy sword also no longer has guaranteed energy projectile deflection; only doing so on a successful block (75% chance to block).
balance: But it does have 40 force now, so it is more lethal a weapon. Traitors can purchase the sword for only 13 TC (down from 16 TC).
balance: The combat energy shield (The one you hold) now functions as a normal shield (it used to only protect you against energy projectiles and nothing else). It loses guaranteed energy projectile deflection, but still reflects the projectile so on a block.
feature: Death commandos continue to have their energy shields deflect all incoming energy projectiles. Because who cares about deathsquads being balanced?
balance: The MODsuit shield module only has one charge, but recharges every 10 seconds. It also costs 8 TC (down from 15). It is also now in the Core Gear beginner box (bringing the total price up to 22 TC).
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
thanks necromanceranne. all these blockchance changes will make people finally use the funny guns that nukie has to offer instead of the same old desword!

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -275,10 +275,20 @@
...
+		var/obj/projectile/our_projectile = hitby
+
+		if(our_projectile.reflectable) //We handle this via IsReflect() instead.
+			final_block_chance = 0
+
+	return ..()
 
 /obj/item/shield/energy/IsReflect()
-	return HAS_TRAIT(src, TRAIT_TRANSFORM_ACTIVE)
+	if(HAS_TRAIT(src, TRAIT_WIELDED) && prob(block_chance))
```
[the remark this answers is not in the document]
Comment (necromanceranne) [reply code/game/objects/items/shields.dm:290]: whoops, thanks for the catch.

**Comment by optimumtact** [comment]
I spent 3 minutes trying to find this PR after people talked about it.

**Comment by Jackal-boop** [comment]
Did you make an issue just to solve it with this pr

**Comment by necromanceranne** [comment]
No, that issue is still unresolved with this pr.

But I will in fact resolve that issue with my own pr if need be.

**Comment by Higgin** [comment]
Great changes.

**Comment by ATH1909** [comment]
deathsquads currently REALLY lean on their energy shields to not instantly die to friendly fire. you might want to make a 0% block + 100% reflect shield variant just for them or something.

**Comment by necromanceranne** [comment]
> deathsquads currently REALLY lean on their energy shields to not instantly die to friendly fire. you might want to make a 0% block + 100% reflect energy shield variant just for them or something.

Good point, I forgot about deathsquaddies.

But who said it had to be 0% block >:)

**Comment by ATH1909** [comment]
> But who said it had to be 0% block >:)

I like deathsquads being on the _cusp_ of mortality instead of "lol no"

like if you're cracked out enough to actually fight them and come out on top, it should be a testament to your skill/insane build instead of to your block rng

**Comment by YuliceTheDruid** [comment]
> I like deathsquads being on the _cusp_ of mortality instead of "lol no"
> 
> like if you're cracked out enough to actually fight them and come out on top, it should be a testament to your skill/insane build instead of to your block rng

baton rushing should not be the meta against a fucking dsquaddie.

**Comment by ATH1909** [comment]
> > I like deathsquads being on the _cusp_ of mortality instead of "lol no"
> > like if you're cracked out enough to actually fight them and come out on top, it should be a testament to your skill/insane build instead of to your block rng
> 
> baton rushing should not be the meta against a fucking dsquaddie.

baton rushing struggles against squads/groups of people, which deathsquaddies should be in

and if they're not then rip bozo lmao

**Comment by akayeb** [comment]
Pr made by anrobust don't merge it or your balls will explode

**Comment by Deultus** [comment]
Firstly, they improve the lasers, the laser protection becomes stronger, since the station does not have normal bolistics. they kill the laser protection. The brilliant work of the "geniuses" the power of spam lasers is even more blunt and deadly.

**Comment by SomeoneYouProbablyKnow** [comment]
Everyone talking about swords and shields but damn the shield module price reduction is about to make it a no-brainer purchase. Only 8 for a recharging 1 hit freeby is kinda nuts. No there's literally no reason not to take it

**Comment by necromanceranne** [comment]
That's entirely fine, I would prefer if they did take it. I'm half tempted to put it in the starter gear but that might start to creep up in price a little too much.

**Comment by SomeoneYouProbablyKnow** [comment]
Yeah that would be better i think. Nothing against it being strong i just don't like stuff you kinda have to buy no matter what.

I wish there was like a system to sell your starter gear at the base. That way you'd be able to really beef up a regular nuckie and still let em customize.

**Comment by SimplySmiley123** [comment]
> > Firstly, they improve the lasers, the laser protection becomes stronger, since the station does not have normal bolistics. they kill the laser protection. The brilliant work of the "geniuses" the power of spam lasers is even more blunt and deadly.
> 
> I mean, there is protection and then there is negation. The current issue, right now, is that a whole lot of guns are not functionally usable against people with deswords or eshields. Barely usable if they have a shielded module of some kind. And so utilizing energy weapons in these scenarios is banking on your target not having these particular bits of equipment or you're almost guaranteed to eat shit. And that isn't a gamble most folk want to take, since those same people are usually difficult to fight in other aspects as well (the desword's 75% block is universal, and shielded module charges will block anything that slips past that, as an example. If they're not using a desword, the module will usually make someone difficult to fight either way).
> 
> So what gets left for the crew to use? The list is slowly shrinking as we push crew weaponry to be energy-based, so the crews options begin to dwindle over time. We still have some pretty decent ballistics around (the new crossbows for example), but something doesn't sit right with me when crew have to make jury-rigged weaponry to combat a paramilitary force coming to kill them rather than requesting more guns through cargo to supplement the crew. The latter option just always feels second-rate, and most folk do not like wanting to use lasers in these situations because they know they are going to meet the inevitable threat that outright nullifies their only weapons. They lost before they had even come to the fight.
> 
> If we want to establish energy weapons as the standard for Nanotrasen, we have to make concessions elsewhere to at least allow that to make logical sense in the game world. Right now, it doesn't. And with that hurdle stepped over, we can more comfortably design future features knowing that we won't have these kinds of problems again. It's a long term perspective, as all these changes have been.

Honestly all these issues with de-swords, e-shields, shield modules and in general, RNG negation mechanics, are a result of axing any and all ballistics for the crew and gimping remaining ballistics to the point where it's pointless to bother. Like seriously, what was expected to happen after polarising crew weaponry when antagonists had ways to completely nullify those weapons one way or another?
RNG blocks are still bad no matter what, RNG shouldn't really dictate who wins, but ballistics and especially shotguns kept it in check.

**Comment by necromanceranne** [comment]
A ballistic gun has no distinct advantage over an energy gun in terms of RNG (except for the specific advantage I've included in the desword, obviously). Even if we're talking volume of projectiles to get through the RNG block faster; at this stage, laser carbines/disabler smgs actually outclass WT-550's. If we're talking pure damage numbers; when buckshot had about 80 force, it existed alongside the taser. It had other rocket tagging firearms to compete with. This is the same historical period that I spoke of in the PR body that justified having the eshield and desword deflect all energy projectiles. So the same root problem.

RNG block is at times an icky mechanic to experience, but I view it as one of the few concessions we just have to deal with due to the limited ways in which players can interface with attacks to begin with, and how it is only as deep as 'I click on the guy to do the thing'. For the most part, it boils down to whoever clicks first and whoever attacks the fastest just to prevent a response. Even avoiding gunfire is a mixture of 'just don't be there when the projectiles are flying' or 'move like a headless chicken so that it is _harder_ to shoot you'.

Whenever we get swing combat will bring about the change we need here, but in the now, it is what it is.

**Comment by Deultus** [comment]
instead of dancing with tambourines, you could just return some of the ballistics to normal, and not "I have to make artisanal cartridges to cause normal damage" and call artisanal crossbows a normal weapon - it's funny, you can't kill anyone with it.

**Comment by Yttriums** [comment]
I don't really see how warops can win against crew without laser negation items. It's pretty trivial for 1-2 competent cargo players to arm everyone with multiple lasers, batons, bulletproof armor and medkits - without dedicated laser counters, that strat is going to be pretty unbeatable

**Comment by moocowswag** [comment]
just keep in mind things like hulk are barred from using desword literally because of the 100% laser blockrate so you are making changes that outdate current balance philosophies and can have an effect on a lot of other things in the future.

**Comment by necromanceranne** [comment]
> I don't really see how warops can win against crew without laser negation items. It's pretty trivial for 1-2 competent cargo players to arm everyone with multiple lasers, batons, bulletproof armor and medkits - without dedicated laser counters, that strat is going to be pretty unbeatable

If this is true, we can work on that I'm sure. If, right now, ops are hinging on energy weapon negation as much as crew are hinging on ballistics to deal with ops such that their win probability plummets, that is a part of that same balancing problem that we can fix over time. But I've my doubts this is the case. I suppose it remains to be seen.

> just keep in mind things like hulk are barred from using desword literally because of the 100% laser blockrate so you are making changes that outdate current balance philosophies and can have an effect on a lot of other things in the future.
> 
> (also this is will be yet another buff to lasers, even if its indirect it goes a long way to making them more powerful)

There are only a few areas of the game I'm not willing to touch, design wise, and one of those things is making any bit of my code touch hulk balance. 

Even if I gave a shit, hulk is so overly politicized that a maint will have me scrap it out anyway. Don't take that personally, it's just a reality of the problem of hulks.

**Comment by Loafin34** [comment]
Great, so now nukies are gonna be forced into finding cover. in completely transparent hallways. where their ballistics are at a disadvantage. If you're gonna do this, make it so that EMPs are more achievable, and have that be a counter to the dumbassery of 3 laser guns. 800 credits from cargo should not consistently drop a nukie.

**Comment by SomeoneYouProbablyKnow** [comment]
You know what. Fuck war ops, if you don't want to be blasted by 3 cargo techs with more budget than the US military consider not pressing the war ops button

**Comment by necromanceranne** [comment]
> Great, so now nukies are gonna be forced into finding cover. in completely transparent hallways. where their ballistics are at a disadvantage. If you're gonna do this, make it so that EMPs are more achievable, and have that be a counter to the dumbassery of 3 laser guns. 800 credits from cargo should not consistently drop a nukie.

Emp kits are 2 TC my guy. You can even use them on demand with the implant. You can shit emps like no tomorrow.

**Comment by Loafin34** [comment]
we are at an impasse now. scarp is legitimately more viable than desword for the first time in years.

**Comment by necromanceranne** [comment]
> Why does every solution to every antag have to be spam lasers until the guy dies? There are already so many other solutions to a desworder to counter them.

I can't believe after years of balance pr'ing I've finally hit a pr where someone made the argument that lasers are _overperforming_ against threats. After all these years, I've finally come full circle.

**Comment by Overuseddart** [comment]
> I can't believe after years of balance pr'ing I've finally hit a pr where someone made the argument that lasers are _overperforming_ against threats. I've finally come full circle.

Because they are kinda the laser buff means that they shred through antags with 25 damage lasers, every antag has been feeling the effect of the change, it’s much harder to fight security now. It doesn’t help that it’s going to be used every round too. 1-2 lasers causes slowdown and constant tox damage from wounds which will kill antags if they don’t have an escape method and heals.

**Comment by Higgin** [comment]
> Because they are the laser buff means that they shred through antags with 25 damage lasers every antag has been feeling the effect of the change and it’s much harder to fight security now. 1-2 lasers causes slowdown and constant tox damage from wounds which will kill antags if they don’t have an escape method and heals.

"getting hit while I am not able to stop fighting and heal" is a fine reason to die. Very few people die of burn wounds. You have to wait a hot fucking minute of not literally just meshing and gauzing yourself to upgrade a burn.

Antags have a lot of escapes or should be planning with them unless they plan to get an elite MOD, stim, and go apeshit. Even just regular armor (which has better energy resist btw) and noslips are a good, accessible combo to go toe-to-toe with lasers if wounds are your concern.

Wounds exist as a modifier and limiter on the combat system. Full immunity to playing around the very weakest of them and basically all of the crew's ranged options for one pickup you can fit in your backpack isn't good design - it's not even with the tradeoff of something like Scarp, which makes it its whole deal that this is going to be "entirely your thing," and still requires you to manage keeping your hands empty, combat)throw mode - etc.

Frankly, skill issue. Until active blocks, 
unreliable passives are still much better than perfect passives.

**Comment by Overuseddart** [comment]
> "getting hit while I am not able to stop fighting and heal" is a fine reason to die. Very few people die of burn wounds. You have to wait a hot fucking minute of not literally just meshing and gauzing yourself to upgrade a burn.

Nukies can’t heal burn wounds without taking off their armor (death sentence when you’re found out or war ops)

> Antags have a lot of escapes or should be planning with them unless they plan to get an elite MOD, stim, and go apeshit. Even just regular armor (which has better energy resist btw) and noslips are a good, accessible combo to go toe-to-toe with lasers if wounds are your concern.
The concern is slowdown and wounds making lasers literally the best option for every antag
Heretic - Shoot em
Blob - Shoot em
Nukies - (Now) shoot em
Tots - Shoot em
Wizards - Shoot em 
Full immunity to playing around the very weakest of them and basically all of the crew's ranged options for one pickup you can fit in your beltslot isn't good design - it's not even comparable with the tradeoff of something like Scarp, which makes it its whole deal that this is going to be "entirely your thing," and still requires you to manage keeping your hands empty, combat)throw mode - etc. Scarp is much more an active block than desword.

You don’t have to have your hands empty for scarp and also scarp needs no weapons to use. You could just tackle or body throw a desworder to steal it leaving them with no weapon and just kill them. You can use the crews biggest advantage  - Numbers! In order to swarm the antag get batong hit in or just body throw each other.

**Comment by Admiral-Gigaton** [comment]
cant wait for the "arm implanter" item that some coder demon implements, and it destabilizes literally everything combat has had changed about it

**Comment by SomeoneYouProbablyKnow** [comment]
> I've reduced the price of the double-bladed energy sword for traitors down to 13 TC

Reasonable but doesn't it make it too similar to the single bladed e sword in price? Although i donno what to do about that.

**Comment by necromanceranne** [comment]
> > I've reduced the price of the double-bladed energy sword for traitors down to 13 TC
> 
> Reasonable but doesn't it make it too similar to the single bladed e sword in price? Although i donno what to do about that.

_shrug_

**Comment by ATH1909** [comment]
> > Nukies can’t heal burn wounds without taking off their armor (death sentence when you’re found out or war ops)
> 
> Yes you can. Patches go through suits, and the combat injector pierces suits as well.

wait, do you mean regenerative mesh? are you saying that that can heal burn wounds through hardsuits?

if not, chem patches do go through hardsuits, but to my awareness, they don't cure burn wounds, which will eventually* poison you. I think that's what this poster was concerned about.

**Comment by necromanceranne** [comment]
> wait, do you mean regenerative mesh? are you saying that that can heal burn wounds through hardsuits?
> 
> if not, chem patches do go through hardsuits, but to my awareness, they don't cure burn wounds, which will eventually* poison you. I think that's what this poster was concerned about.

Ah, if that was what they meant, then yeah regen mesh needs you to remove your suit. However, infection on low level wounds takes a good while to set in. You might just be able to win before then. 

And, in all honesty, I've personally never been burned through an elite hardsuit by a standard laser. I don't think that's even anecdotal evidence either.

Lemme check the code real quick. Syndicate jumpsuits provide 10 wound armor, and the blood-red/elite hardsuit also provides 25 wound armor on its own. so -35 wounding power for most nukies. Blood-red has 30 laser armor while armor boosted, and elite has 50. 

Normal lasers have 25 force, but -20 wounding power. Wounding bonuses are applied after damage is reduced by armor and the wounding exponential (x1.4) is applied to the resulting damage value for the sake of determining the wound roll value. For the most part, our armors will be putting the damage value below 20. So we'll be rolling with a negative value (this means we don't get wounded ever). Wounding armor is a direct penalty to a potential wound roll. so any given value, we penalize that by another -35.

I think it is statically impossible to wound someone in either of these suits with a laser, even in the most ideal circumstances, because nothing is able to reach a wounding threshold. So I think this is a non-issue.

**Comment by optimumtact** [comment]
I'm okay with lasers being powerful for a while they have been a joke for a long time we can see how it goes.

**Comment by Ghommie** [comment]
I'm not entirely sold on some things, but I'm willing to see how it goes, especially considering how tackling is a high-risk move.

**Pull request merged**